### PR TITLE
fix/google-calendar-removes-attendees-when-seats-5336

### LINKED ISSUE #5336: [CAL-307] Google calendar connection sends cancellation emails as new seats are booked
When a second seat is booked the first seat receives an email informing them of the event cancellation. This happens when the organizer has a google calendar connected.

This also leaks additional notes of the second seat as part of the cancellation email.

> From [SyncLinear.com](https://synclinear.com)

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

Fixes #5336 

**Environment**: Staging(main branch)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Update an event type with seats available
- [x] Install google calendar in that event type
- [x] Book that event multiple times
- [x] Users already booked shouldn't been remove from calendar Event thus they won't receive an email about event cancellation.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/components/v2/eventtype/EventAdvancedTab.tsx [hunk 1]
```diff
@@ -329,9 +326,10 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupInfered
                     label={t("number_of_seats")}
                     type="number"
                     defaultValue={value || 2}
+                    min={1}
                     addOnSuffix={<>{t("seats")}</>}
                     onChange={(e) => {
-                      onChange(Number(e.target.value));
+                      onChange(Math.abs(Number(e.target.value)));
```
Comment (alannnc) [inline apps/web/components/v2/eventtype/EventAdvancedTab.tsx:332]: NIT improvement to don't allow negative numbers on input field. (Should validate on form and server too probably)

File: packages/app-store/googlecalendar/lib/CalendarService.ts [hunk 2]
```diff
@@ -160,6 +160,10 @@ export default class GoogleCalendarService implements Calendar {
   async updateEvent(uid: string, event: CalendarEvent, externalCalendarId: string): Promise<any> {
     return new Promise(async (resolve, reject) => {
       const myGoogleAuth = await this.auth.getToken();
+      const eventAttendees = event.attendees.map(({ id, ...rest }) => ({
+        ...rest,
+        responseStatus: "accepted",
+      }));
```
Comment (alannnc) [inline packages/app-store/googlecalendar/lib/CalendarService.ts:166]: Just clean up

File: packages/core/EventManager.ts [hunk 3]
```diff
@@ -294,6 +294,13 @@ export default class EventManager {
   }
 
   public async updateCalendarAttendees(event: CalendarEvent, booking: PartialBooking) {
+    const attendees = booking.attendees.map((attendee) => {
+      return {
+        email: attendee.email,
+        name: attendee.name,
+      };
+    });
+    event.attendees = attendees;
```
Comment (alannnc) [inline packages/core/EventManager.ts:303]: This was the problem, it looks like code that copies attendees to event was missing here.

File: packages/core/EventManager.ts [hunk 4]
```diff
@@ -430,11 +437,14 @@ export default class EventManager {
     try {
       // Bookings should only have one calendar reference
       calendarReference = booking.references.filter((reference) => reference.type.includes("_calendar"))[0];
-      if (!calendarReference) throw new Error("bookingRef");
-
+      if (!calendarReference) {
+        throw new Error("bookingRef");
+      }
```
Comment (alannnc) [inline packages/core/EventManager.ts:442]: Clean up like the rest of the code below this change

File: packages/core/EventManager.ts (hunk 3 above)
Comment (emrysal) [reply packages/core/EventManager.ts:312]: Is there any situation where event.attendees would be set? (And this would overwrite that existing value?)

**Review by emrysal — CHANGES_REQUESTED** [review]
We've got to figure out why https://github.com/calcom/cal.com/blob/352ce35b40cc68b3d7830619198e78e58946c259/packages/features/bookings/lib/handleNewBooking.ts#L485 does not work.

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 5]
```diff
@@ -511,8 +517,13 @@ async function handler(req: NextApiRequest & { userId?: number | undefined }) {
...
     const newSeat = booking.attendees.length !== 0;
-
-    await sendScheduledSeatsEmails(evt, invitee[0], newSeat, !!eventType.seatsShowAttendees);
+    /**
+     * Remember objects are passed into functions as references
+     * so if you modify it in a inner function it will be modified in the outer function
+     * deep cloning evt to avoid this
+     */
+    const copyEvent = cloneDeep(evt);
+    await sendScheduledSeatsEmails(copyEvent, invitee[0], newSeat, !!eventType.seatsShowAttendees);
```
Comment (alannnc) [inline packages/features/bookings/lib/handleNewBooking.ts:526]: @emrysal Turns out this function was modifying attendees due to using the object as reference. Now using a deepCloned object to send the email and updateEvent with the original event.

**Review by emrysal — APPROVED** [review]

**Pull request merged**